src/Symfony/Component/Security/Core/Authorization/Voter/Vote.php

```diff
@@ -28,6 +28,8 @@ class Vote
      */
     public array $reasons = [];
 
+    public array|object|null $extraData = null;
```

Comment:
I would only accept an array, to encourage ppl that use this to properly use keys as namespaces:
```suggestion
    public array $extraData = [];
```